datafusion/expr-common/src/signature.rs

```diff
@@ -105,6 +105,7 @@ pub enum TypeSignature {
     Uniform(usize, Vec<DataType>),
     /// Exact number of arguments of an exact type
     Exact(Vec<DataType>),
+    Coercible(Vec<DataType>),
```

Comment:
perhaps we could add some comments here explaining what Coercible meant? Maybe it means that the input arguments can be coerced to one of the target types (in order)?

Reply:
Reading this again, I am not sure how this is different than `Exact(Vec<DataType>)` as I thought the coercion logic would try to coerce the arguments to the types in `Exact` 🤔  

Is the difference is that the input isn't actually converted to the `Coercable` types by DataFusion, but instead the function's implementation handles coercion internally?